src/Bonnier/WP/Cxense/Services/DocumentSearch.php

```diff
@@ -0,0 +1,325 @@
+<?php
+/**
+ * DocumentSearch file class for cxense api
+ */
+
+namespace Bonnier\WP\Cxense\Services;
+
+use Bonnier\WP\Cxense\Exceptions\DocumentSearchMissingSearch;
+use Bonnier\WP\Cxense\Exceptions\DocumentSearchMissingCount;
+use Bonnier\WP\Cxense\Exceptions\DocumentSearchMissingFacet;
+use Bonnier\WP\Cxense\Exceptions\DocumentSearchWrongFilter;
+use Bonnier\WP\Cxense\Http\HttpRequest;
+use Bonnier\WP\Cxense\Settings\SettingsPage;
+use Bonnier\WP\Cxense\Parsers\Document;
+use Bonnier\WP\Cxense\Parsers\Facet;
+
+/**
+ * DocumentSearch class
+ */
+class DocumentSearch {
+	
+	/**
+	 * Instance object
+	 *
+	 * @var DocumentSearch $objInstance
+	 */
+	private static $objInstance;
+	
+	/**
+	 * Search array
+	 *
+	 * @var array $arrSearch
+	 */
+	private $arrSearch = [];
+	
+	/**
+	 * Payload array
+	 *
+	 * @var array $arrPayload
+	 */
+	private $arrPayload = [];
+
+	/**
+	 * Settings object
+	 *
+	 * @var SettingsPage $objSettings
+	 */
+	private $objSettings;
+	
+	/**
+	 * Returning fields
+	 *
+	 * @var array $arrFields
+	 */
+	private $arrFields = [
+		'bod-cat',
```

Comment:
Can you check this one out: https://github.com/BenjaminMedia/wp-cxense/pull/7/commits/53c2343bb92bca7a0535e0f510c4f189ebcb8882 ?